staging/src/k8s.io/client-go/tools/portforward/portforward.go

```diff
@@ -197,8 +197,9 @@ func (pf *PortForwarder) forward() error {
 	var err error
 
 	listenSuccess := false
-	for _, port := range pf.ports {
-		err = pf.listenOnPort(&port)
+	for i, _ := range pf.ports {
+		port := &pf.ports[i]
+		err = pf.listenOnPort(port)
```

Comment:
My concern was that listenOnPort would modify the port in-place (and then this code would be a behavioural change), but that does not appear to be the case. :+1:

Reply:
Wondering if we should change `listenOnPort` to just take a ForwardedPort, rather than `*ForwardedPort`, given that it is so small anyway

Reply:
@justinsb What did you have in mind? One option might be to have `getListener()` (maybe also with an appropriate rename, e.g. to `buildListener()`) also return a ForwardedPort containing the correct local port, which is passed up through the call stack to `forward()` where it's used to replace the entry in `pf.ports` or build a new `pr.ports` from scratch.

Reply:
`listenOnPort` sets the `port.Local` if a local port was 0. It's intended. The behavior change is a bug fix.